src/pages/api/search/intent.ts

```diff
@@ -53,7 +59,16 @@ const handler: NextApiHandler = async (req, res) => {
       return res.status(200).json(payload);
     }
 
-    const payload = searchIntentSchema.parse(DUMMY_AI_RESPONSE);
+    const modelResponse = await callModel(searchTerm);
```

Comment:
The name `modelResponse` implies to me that its a structured response of some kind. Looking at the  `modelResponse.map` below, it seems to just be a list of subjects, in which case we could call it something like `subjectsFromModel`